src/types/resolveDescriptors.ts

```diff
@@ -1552,6 +1546,17 @@ export function resolveDescriptors(ctx: CompilerContext, Ast: FactoryAst) {
                         `Trait "${tr.name}" requires field "${f.name}" of type "${printTypeRef(f.type)}"`,
                         t.ast.loc,
                     );
+                } else if (f.as !== null && f.as !== ex.as) {
+                    const expected = `${printTypeRef(f.type)} as ${f.as}`;
+                    let actual = printTypeRef(ex.type);
+                    if (ex.as !== null) {
+                        actual += ` as ${ex.as}`;
+                    }
```

Comment:
let's use a conditional expression here and remove mutation, please check our code style guide: https://github.com/tact-lang/tact/blob/main/STYLEGUIDE.md

Reply:
fixed